cmake/adjust_global_compile_flags.cmake

```diff
@@ -373,4 +373,6 @@ if (WIN32)
     string(APPEND CMAKE_CXX_FLAGS " /MP")
     # required to be set explicitly to enable Eigen-Unsupported SpecialFunctions
     string(APPEND CMAKE_CXX_FLAGS " -DEIGEN_HAS_C99_MATH")
+elseif(NOT ${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
+    add_compile_definitions("_GNU_SOURCE")
```

Comment:
Might not be needed. Cmake has an option to allow us switching between C++17 and gnu++17. The later implies "_GNU_SOURCE".

This change has impact on who calls strerror_r directly as the function's signature is different when _GNU_SOURCE is defined or not.